tests/icon-api.js

```diff
@@ -0,0 +1,252 @@
+  t.alike(icon, expectedSmallIcon)
+})
+
+test(`icon api - create and fail to find variant with matching mimeType`, async (t) => {
+  // const iconDoc = await iconApi[kCreate]({
+  //   name: 'myIcon',
+  //   schemaName: 'icon',
+  //   variants: [
+  //     {
+  //       size: 'small',
+  //       pixelDensity: 1,
+  //       mimeType: 'image/svg+xml',
+  //       blob: Buffer.from(expectedSmallIcon),
+  //     },
+  //   ],
+  // })
+
+  // t.exception(
+  //   await iconApi[kGetIcon]({
+  //     iconId: iconDoc.docId,
+  //     mimeType: 'image/png',
+  //   }),
+  //   'no matching mimeType for icon'
+  // )
+  t.pass()
+})
+
+test('icon api - create and get with different variants', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+      {
+        size: 'medium',
+        pixelDensity: 1,
+        mimeType: 'image/svg+xml',
+        blob: Buffer.from(expectedMediumIcon),
+      },
+      {
+        size: 'large',
+        pixelDensity: 2,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedLargeIcon),
+      },
+    ],
+  })
+
+  const { icon } = await iconApi[kGetIcon]({
+    iconId: iconDoc.docId,
+    size: 'large',
+  })
+  t.alike(icon, expectedLargeIcon)
+})
+
+test('icon api - create and get with variants, choosing the variant with more matching criteria', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+      {
+        size: 'medium',
+        pixelDensity: 1,
+        mimeType: 'image/svg+xml',
+        blob: Buffer.from(expectedMediumIcon),
+      },
+      {
+        size: 'large',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedLargeIcon),
+      },
+    ],
+  })
+
+  const { icon } = await iconApi[kGetIcon]({
+    iconId: iconDoc.docId,
+    size: 'large',
+    pixelDensity: 1,
+    mimeType: 'image/png',
+  })
+  t.alike(icon, expectedLargeIcon)
+})
+
+test('icon api - create and get with variants, choosing the first variant with the first best score', async (t) => {
+  const iconDoc = await iconApi[kCreate]({
+    name: 'myIcon',
+    schemaName: 'icon',
+    variants: [
+      {
+        size: 'small',
+        pixelDensity: 1,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedSmallIcon),
+      },
+      {
+        size: 'medium',
+        pixelDensity: 1,
+        mimeType: 'image/svg+xml',
+        blob: Buffer.from(expectedMediumIcon),
+      },
+      {
+        size: 'large',
+        pixelDensity: 2,
+        mimeType: 'image/png',
+        blob: Buffer.from(expectedLargeIcon),
```

Comment:
Can remove unnecessary conversions:

```suggestion
        blob: expectedSmallIcon,
      },
      {
        size: 'medium',
        pixelDensity: 1,
        mimeType: 'image/svg+xml',
        blob: expectedMediumIcon,
      },
      {
        size: 'large',
        pixelDensity: 2,
        mimeType: 'image/png',
        blob: expectedLargeIcon,
```